### PR TITLE
Variant type storage added

### PR DESCRIPTION
**Pull request by carolyncaron**
## Metadata

 - Issue #21 

- [x] I feel this PR is ready to be merged.
- [ ] This PR is dependent upon [PR #/ nothing]

Documentation:
- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.

## Description

Previously, there was no featureprop record for variant types which made this module incompatible with [ND Genotypes](https://github.com/UofS-Pulse-Binfo/nd_genotypes/). So in addition to implementing that functionality, I also simplified the way in which cvterms for data types were extracted and added the option to select the cvterm for variant type to the settings form.

## Dependencies

While not dependent necessarily, it is worth mentioning that Issue #15 should be tackled next to provide the user with more flexibility in selecting cvterms, as this implementation reinforces the strict requirement that variant type be selected from the feature_property CV only.

## Testing?

See PR #18 for detailed instructions on setting up this module in a cloud9 environment.

- [x] I tested on a generic Tripal Site
- [ ] I tested on a KnowPulse Clone
- [ ] This PR includes automated testing

### CODE REVIEW

File: theme/js/adminCheckCVTerms.js [hunk 1]
```diff
@@ -26,9 +26,11 @@ console.log('initial check.');
       var value = element.attr("value");
       var url = autocomplete_url_prefix + cv_id + '/' + value;
 
+      console.log(url);
       // Grab the JSON found using the autocomplete callback...
       $.getJSON(url, function( data ) {
-
+        
+        console.log(data);
```
Comment (laceysanderson) [inline theme/js/adminCheckCVTerms.js]: You forgot a couple logging statements:

```suggestion
```

File: theme/js/adminCheckCVTerms.js [hunk 2]
```diff
@@ -26,9 +26,11 @@ console.log('initial check.');
       var value = element.attr("value");
       var url = autocomplete_url_prefix + cv_id + '/' + value;
 
+      console.log(url);
```
Comment (laceysanderson) [inline theme/js/adminCheckCVTerms.js]:
```suggestion
```

File: theme/js/adminCheckCVTerms.js [hunk 3]
```diff
@@ -9,7 +9,7 @@
   Drupal.behaviors.adminCheckCVTerms = {
     attach: function (context, settings) {
 
-    var elementNames = ['sample_type', 'stock_rel', 'feature_rel', 'featureprop_type'];
+    var elementNames = ['sample_type', 'stock_rel', 'feature_rel', 'featureprop_type_marker','featureprop_type_variant'];
 console.log('initial check.');
```
Comment (laceysanderson) [inline theme/js/adminCheckCVTerms.js:13]:
```suggestion
```

**Review by laceysanderson — APPROVED** [review]
Code review 👍

**Review by laceysanderson — APPROVED** [review]
2nd Code review looks good 👍

File: includes/genotypes_loader.admin.inc [hunk 4]
```diff
@@ -141,11 +141,21 @@ function genotypes_loader_admin_settings_form($form, &$form_state) {
   );
 
   $feature_prop_cv = tripal_get_cv(array('name' => 'feature_property'));
-  $form['cvterms']['featureprop_type'] = array(
+  $form['cvterms']['featureprop_type_marker'] = array(
     '#type' => 'textfield',
     '#title' => 'Feature Property: marker_type',
     '#description' => 'This cvterm is used as the featureprop.type_id for the property indicating the type of marker.',
-    '#default_value' => variable_get('genotypes_featureprop_type', 'marker_type'),
+    '#default_value' => variable_get('genotypes_featureprop_type_marker', 'marker_type'),
```
Comment (laceysanderson) [inline includes/genotypes_loader.admin.inc]:
```suggestion
    '#default_value' => variable_get('genotypes_featureprop_type_marker', 'additionalType'),
```

File: includes/genotypes_loader.admin.inc [hunk 5]
```diff
@@ -141,11 +141,21 @@ function genotypes_loader_admin_settings_form($form, &$form_state) {
...
+    '#autocomplete_path' => 'ajax/genotypes_loader/cvterm/' . $feature_prop_cv->cv_id,
+    '#attributes' => array('data-cv' => 'feature_property', 'data-cv_id' => $feature_prop_cv->cv_id ),
+  );
+
+  $feature_prop_cv = tripal_get_cv(array('name' => 'feature_property'));
+  $form['cvterms']['featureprop_type_variant'] = array(
+    '#type' => 'textfield',
+    '#title' => 'Feature Property: variant_type',
+    '#description' => 'This cvterm is used as the featureprop.type_id for the property indicating the type of variant.',
+    '#default_value' => variable_get('genotypes_featureprop_type_variant', 'variant_type'),
```
Comment (laceysanderson) [inline includes/genotypes_loader.admin.inc]:
```suggestion
    '#default_value' => variable_get('genotypes_featureprop_type_variant', 'additionalType'),
```

File: README.md [hunk 6]
```diff
@@ -119,7 +119,8 @@ This module currently expects the following controlled vocabulary terms already
 | Sample => germplasm relationship | is_extracted_from   | stock_relationship    |
 | Marker => variant relationship | is_marker_of        | stock_relationship    |
 | Marker type | *Indicated by user* | sequence              |
-| Variant type | *Indicated by user* | sequence              |
 | Property type for free-text marker description | marker_type         | feature_property      |
+| Variant type | *Indicated by user* | sequence              |
+| Property type for free-text variant description | variant_type        | feature_property      |
```
Comment (laceysanderson) [inline README.md]:
```suggestion
| Property type for free-text variant description | additionalType        | schema      |
```

File: README.md [hunk 7]
```diff
@@ -119,7 +119,8 @@ This module currently expects the following controlled vocabulary terms already
 | Sample => germplasm relationship | is_extracted_from   | stock_relationship    |
 | Marker => variant relationship | is_marker_of        | stock_relationship    |
 | Marker type | *Indicated by user* | sequence              |
-| Variant type | *Indicated by user* | sequence              |
 | Property type for free-text marker description | marker_type         | feature_property      |
```
Comment (laceysanderson) [inline README.md]:
```suggestion
| Property type for free-text marker description | additionalType         | schema      |
```

**Review by laceysanderson — CHANGES_REQUESTED** [review]
There should be two options for marker/variant type: SO term and free text description.

**Review by laceysanderson — APPROVED** [review]

File: includes/genotypes_loader.admin.inc [hunk 8]
```diff
@@ -141,11 +141,21 @@ function genotypes_loader_admin_settings_form($form, &$form_state) {
   );
 
   $feature_prop_cv = tripal_get_cv(array('name' => 'feature_property'));
```
Comment (carolyncaron) [inline includes/genotypes_loader.admin.inc]:
```suggestion
  $feature_prop_cv = tripal_get_cv(array('name' => 'schema'));
```

File: includes/genotypes_loader.admin.inc [hunk 9]
```diff
@@ -141,11 +141,21 @@ function genotypes_loader_admin_settings_form($form, &$form_state) {
...
-  $form['cvterms']['featureprop_type'] = array(
+  $form['cvterms']['featureprop_type_marker'] = array(
     '#type' => 'textfield',
     '#title' => 'Feature Property: marker_type',
     '#description' => 'This cvterm is used as the featureprop.type_id for the property indicating the type of marker.',
-    '#default_value' => variable_get('genotypes_featureprop_type', 'marker_type'),
+    '#default_value' => variable_get('genotypes_featureprop_type_marker', 'additionalType'),
+    '#autocomplete_path' => 'ajax/genotypes_loader/cvterm/' . $feature_prop_cv->cv_id,
+    '#attributes' => array('data-cv' => 'feature_property', 'data-cv_id' => $feature_prop_cv->cv_id ),
```
Comment (carolyncaron) [inline includes/genotypes_loader.admin.inc]:
```suggestion
    '#attributes' => array('data-cv' => 'schema', 'data-cv_id' => $feature_prop_cv->cv_id ),
```

File: includes/genotypes_loader.admin.inc [hunk 10]
```diff
@@ -141,11 +141,21 @@ function genotypes_loader_admin_settings_form($form, &$form_state) {
...
+  );
+
+  $feature_prop_cv = tripal_get_cv(array('name' => 'feature_property'));
```
Comment (carolyncaron) [inline includes/genotypes_loader.admin.inc]:
```suggestion
  $feature_prop_cv = tripal_get_cv(array('name' => 'schema'));
```

File: includes/genotypes_loader.admin.inc [hunk 11]
```diff
@@ -141,11 +141,21 @@ function genotypes_loader_admin_settings_form($form, &$form_state) {
...
+  $form['cvterms']['featureprop_type_variant'] = array(
+    '#type' => 'textfield',
+    '#title' => 'Feature Property: variant_type',
+    '#description' => 'This cvterm is used as the featureprop.type_id for the property indicating the type of variant.',
+    '#default_value' => variable_get('genotypes_featureprop_type_variant', 'additionalType'),
     '#autocomplete_path' => 'ajax/genotypes_loader/cvterm/' . $feature_prop_cv->cv_id,
     '#attributes' => array('data-cv' => 'feature_property', 'data-cv_id' => $feature_prop_cv->cv_id ),
```
Comment (carolyncaron) [inline includes/genotypes_loader.admin.inc]:
```suggestion
    '#attributes' => array('data-cv' => 'schema', 'data-cv_id' => $feature_prop_cv->cv_id ),
```

**Pull request merged**